gordo_components/builder/build_model.py

```diff
@@ -32,593 +33,577 @@
+                model = None
+                self.metadata["model"] = {
+                    "cross-validation": {
+                        "cv-duration-sec": cv_duration_sec,
+                        "scores": scores,
+                    }
                 }
-                val.update(
+                return model, self.metadata
+        # Train
+        logger.debug("Starting to train model.")
+        start = time.time()
+        model.fit(X, y)
+        time_elapsed_model = time.time() - start
+
+        metadata: Dict[Any, Any]
+        metadata = {"user-defined": self.metadata}
+        metadata["name"] = self.name
+        metadata["dataset"] = dataset.get_metadata()
+        utc_dt = datetime.datetime.now(datetime.timezone.utc)
+        metadata["model"] = {
+            "model-offset": self._determine_offset(model, X),
+            "model-creation-date": str(utc_dt.astimezone()),
+            "model-builder-version": __version__,
+            "model-config": self.model_config,
+            "data-query-duration-sec": time_elapsed_data,
+            "model-training-duration-sec": time_elapsed_model,
+            "cross-validation": {"cv-duration-sec": cv_duration_sec, "scores": scores},
+        }
+
+        metadata["model"].update(self._extract_metadata_from_model(model))
+
+        return model, metadata
+
+    def set_seed(self, seed: int):
+        logger.info(f"Setting random seed: '{seed}'")
+        tf.random.set_seed(seed)
+        np.random.seed(seed)
+        random.seed(seed)
+
+    @staticmethod
+    def build_metrics_dict(
+        metrics_list: list,
+        y: pd.DataFrame,
+        scaler: Optional[Union[TransformerMixin, str]] = None,
+    ) -> dict:
+
+        """
+        Given a list of metrics that accept a true_y and pred_y as inputs this returns a
+        dictionary with keys in the form '{score}-{tag_name}' for each given target tag
+        and '{score}' for the average score across all target tags and folds,
+        and values being the callable make_scorer(metric_wrapper(score)). Note: score in
+        {score}-{tag_name} is a sklearn's score function name with '_' replaced by '-'
+        and tag_name corresponds to given target tag name with ' ' replaced by '-'.
+
+        Parameters
+        ----------
+        metrics_list: list
+            List of sklearn score functions
+        y: pd.DataFrame
+            Target data
+        scaler : Optional[Union[TransformerMixin, str]]
+            Scaler which will be fitted on y, and used to transform the data before
+            scoring. Useful when the metrics are sensitive to the amplitude of the data, and
+            you have multiple targets.
+
+
+        Returns
+        -------
+            dict
+        """
+        if isinstance(scaler, str):
+            scaler = serializer.pipeline_from_definition(scaler)
+
+        if scaler:
+            logger.debug("Fitting scaler for scoring purpose")
+            scaler.fit(y)
```

Comment:
Better to check if it is a str if it actually exists?
```
if scaler:
    logger.debug("Fitting scaler for scoring purpose")
    if isinstance(scaler, str): 
       scaler = serializer.pipeline_from_definition(scaler)
    scaler.fit(y)
```

Reply:
meh, I _think_ it's functionally equivalent, because it'll at least be defined as `None` but I do like the suggestion as it's more concise and arguably more readable, as dealing the scaler logic is in a single place. :+1:

Reply:
Cool, I agree that it is functionally equivalent, and probably not computational heavy to check if it is a string. :sunglasses: